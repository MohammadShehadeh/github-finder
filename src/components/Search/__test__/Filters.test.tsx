import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Filters } from '..';

const mockFilterProps = {
  filters: [
    {
      id: 'users',
      label: 'Users',
    },
    {
      id: 'repositories',
      label: 'Repositories',
    },
  ],
  onSelect: jest.fn(),
  selected: 'users',
};

describe('Filters', () => {
  it('should render to match snapshot', () => {
    const tree = renderer.create(<Filters {...mockFilterProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle filters click correctly', () => {
    render(<Filters {...mockFilterProps} />);

    const button = screen.getAllByRole('button')[1];
    fireEvent.click(button);
    expect(mockFilterProps.onSelect).toHaveBeenCalled();
  });
});
