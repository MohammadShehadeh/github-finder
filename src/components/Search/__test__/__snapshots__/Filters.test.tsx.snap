// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should render to match snapshot 1`] = `
<div
  className="filters"
>
  <p>
    Search type:
  </p>
  <button
    className="button contained small"
    onClick={[Function]}
  >
    Users
  </button>
  <button
    className="button outlined small"
    onClick={[Function]}
  >
    Repositories
  </button>
</div>
`;
